src/main/java/com/alipay/remoting/ConnectionEventListener.java

```diff
@@ -56,7 +56,7 @@ public void addConnectionEventProcessor(ConnectionEventType type,
                                             ConnectionEventProcessor processor) {
         List<ConnectionEventProcessor> processorList = this.processors.get(type);
         if (processorList == null) {
-            this.processors.putIfAbsent(type, new ArrayList<ConnectionEventProcessor>(1));
+            this.processors.putIfAbsent(type, new CopyOnWriteArrayList<ConnectionEventProcessor>());
```

Comment:
I think there need some unit test to reproduce the concurrency problems to check if this modify fixes them .

Reply:
done